Add a benchmark based on `python -m pprint`

arhadthedev opened the pull request:
1. The moved code measures performance so it should belong here.
2. The relocation lifts off a question [whether `python -m pprint` should have tests or not](https://github.com/python/cpython/pull/92560#discussion_r868435161).

After merging, https://github.com/python/cpython/issues/92546 needs to be closed.

ericsnowcurrently reviewed (changes requested):
Thanks for working on this, Oleg!  I've left some comments for you to consider.

arhadthedev commented on pyperformance/data-files/benchmarks/bm_from_stdlib/pyproject.toml:
```diff
@@ -0,0 +1,9 @@
+dynamic = ["version"]
+
+[tool.pyperformance]
+name = "from_stdlib"
```
> What did you have in mind relative to stdlib benchmarks

Initially I've thought that a benchmark should be big, like `Lib/test/*.py` test files.

arhadthedev commented:
@ericsnowcurrently Thank you for your thorough feedback, I've addressed everything.

In addition, I've also moved a class docstring into a file comment, and reworded it together with authorship. I hope I've made it right.

Also I've added `hasattr`-based conditional running of `_safe_repr` to not fail CI for pre-3.10 CPython.

> What did you have in mind relative to stdlib benchmarks? Note that we already have a bunch of micro-benchmarks for various parts of the stdlib, in distinct benchmarks.

Initially I've thought that a benchmark should be big, like `Lib/test/*.py` test files.

ericsnowcurrently reviewed (approved):
Thanks for the updates.
